task/inspect-image/0.1/inspect-image.yaml

```diff
@@ -70,24 +70,49 @@ spec:
       if [[ "${IMAGE_URL}" == *":"*"@"* ]]; then
         IMAGE_URL="${IMAGE_URL/:*@/@}"
       fi
-      echo "Inspecting manifest for source image ${IMAGE_URL}."
-      skopeo inspect --no-tags docker://"${IMAGE_URL}" > $IMAGE_INSPECT 2> stderr.txt || true
-      skopeo inspect --no-tags --raw docker://"${IMAGE_URL}" > $RAW_IMAGE_INSPECT 2>> stderr.txt || true
 
-      if [ ! -z $(cat stderr.txt) ]; then
-        echo "Skopeo inspect failed, inspect-image test encountered the following error:"
-        cat stderr.txt
-        note="Task $(context.task.name) failed: Encountered errors while inspecting image. For details, check Tekton task log."
-        TEST_OUTPUT=$(make_result_json -r ERROR -t "$note")
-        echo "${TEST_OUTPUT}" | tee $(results.TEST_OUTPUT.path)
-        exit 0
+      status=-1
+      max_run=5
+      sleep_sec=10
+      for run in $(seq 1 $max_run); do
+        [ "$run" -gt 1 ] && sleep $sleep_sec  # skip last sleep
+        echo "Inspecting manifest for source image ${IMAGE_URL} (try $run/$max_run)."
+        $(skopeo inspect --no-tags docker://"${IMAGE_URL}" > $IMAGE_INSPECT)
+        status=$?
+        if [ "$status" -eq 0 ]; then
+          break
+        fi
+      done
+      if [ "$status" -ne 0 ]; then
+          # image may not exist, we cannot recover from this error, return exit error code
+          echo "Failed to inspect image ${IMAGE_URL}"
+          exit 1
```

Comment:
This does conflict with [ADR 14. to let pipelines proceed](https://redhat-appstudio.github.io/book/ADR/0014-let-pipelines-proceed.html) but I can definitely see the point of not being able to inspect the image being a breaking issue. 
I still think we'll need to discuss it some more with the other RHTAP developers before committing to a change like this.

Reply:
I can change it to never-fail, but it gives no sense to me, if image cannot be pulled, all subsequent tasks will fail anyway.

ADR 14 says it about `linting and testing` tasks. IMHO this is neither, it's just task that fetches metadata.

Reply:
Yes, but the linting and testing tasks depend on this image inspection task. I think that the spirit of ADR 14 is to allow for users to configure their Enterprise Contract to not require the built-in linting/testing/inspection.
Failing the inspect task would then block their entire component build irrespective of what EC they have configured.

Reply:
okay, I'll update it to neverfailing task

Reply:
updated

Reply:
Thanks. We can follow up with further discussion on ADR 14 in the future